fofix/core/GameEngine.py

```diff
@@ -292,29 +289,94 @@ def __init__(self, config = None):
 
         self.setlistMsg = None
 
+        # Init theme system
+        themename = self.config.get("coffee", "themename")
+        themepath  = os.path.join(Version.dataPath(), "themes", themename)
+        self._initTheme(themename, themepath)
 
         # Load game modifications
         Mod.init(self)
-        self.task.addTask(self.input, synced = False)
 
+        self.task.addTask(self.input, synced = False)
         self.task.addTask(self.view, synced = False)
-
         self.task.addTask(self.resource, synced = False)
 
         self.data = Data(self.resource, self.svg)
 
+        self.input.addKeyListener(FullScreenSwitcher(self), priority = True)
+        self.input.addSystemEventListener(SystemEventHandler(self))
+
+        self.debugLayer         = None
+        self.startupLayer       = None
+        self.loadingScreenShown = False
+        self.graphicMenuShown   = False
+
+        log.debug("Ready.")
+
+
+    # evilynux - This stops the crowd cheers if they're still playing (issue 317).
+    def quit(self):
+        # evilynux - self.audio.close() crashes when we attempt to restart
+        if not self.restartRequested:
+            self.audio.close()
+        Player.savePlayers()
+        for taskData in list(self.task.tasks):
+            self.task.removeTask(taskData['task'])
+        self.running = False
+
+
+    def _initTheme(self, themename, themepath):
+        '''
+        Select the source of graphics for the game.
+        Note that currently this can only be called GameEngine on startup.
+
+        :param themename: what to call this theme
+        :type themename: str
+        :param themepath: absolute path to theme folder
+        :type themepath: str
+        '''
+
+        log.notice( 'Setting theme %s from "%s"' % (themename,themepath) )
+
+        self.theme = None
+        try:
+            # Look for "CustomTheme.py" inside theme dir
+            fp, pathname, description = imp.find_module("CustomTheme",[themepath])
+            try:
+                # Found it! Load it.
+                theme = imp.load_module("CustomTheme", fp, pathname, description)
+                self.theme = theme.CustomTheme(themepath, themename)
+                log.notice('Theme activated using custom class "%s"' % pathname)
+            except ImportError as e:
+                # Unable to load module; log it, but continue with default Theme.
+                log.error('Failed to load CustomTheme.py from "%s"' % pathname)
+            finally:
+                fp.close()
+        except ImportError:
+            # CustomTheme is optional; no need to complain.
```

Comment:
As I said before, please log the error.

Reply:
There is no error; this is the "ask forgiveness" style of programing.  See: https://stackoverflow.com/a/12270384

Note also that the situation is thoroughly logged above and below these lines (the two "Theme activated" lines) so the theme developer knows what happened.

Reply:
If an error is raised, you have to log it. Otherwise, merge the two `try` / `except` blocks as I suggested earlier…
This is even so useful to know why there is an error, and not having to guess it by reading the code.

Reply:
I would just do something along the lines of:
```
log.notice("No CustomTheme.py found in theme %s", themename)
```

Just in-case they were intending on loading a CustomTheme.py and it didn't work for whatever reason...

Reply:
In context, that's not an error.  True, the word "Error" appears in the exception name, but that's just how imp.find_module() communicates with its caller. Since theme are allowed to _not_ have CustomTheme.py files, then it is not an error when find_module() indeed fails to find the file.

But this is your project, and I want my code accepted here, so I'll log the exception.